pkg/natdiscovery/remote_endpoint.go

```diff
@@ -86,17 +102,26 @@ func (rn *remoteEndpointNAT) sinceLastTransition() time.Duration {
 }
 
 func (rn *remoteEndpointNAT) hasTimedOut() bool {
-	return time.Since(rn.started) > toDuration(&totalTimeout)
+	return time.Since(rn.started) > rn.timeout
 }
 
 func (rn *remoteEndpointNAT) useLegacyNATSettings() {
-	rn.useNAT = rn.endpoint.Spec.NATEnabled
-	if rn.endpoint.Spec.NATEnabled {
+	switch {
+	case rn.usingLoadBalancer:
+		rn.useNAT = true
 		rn.useIP = rn.endpoint.Spec.PublicIP
 		rn.transitionToState(selectedPublicIP)
+		klog.V(log.DEBUG).Infof("using NAT for the load balancer backed endpoint %q, using public IP %q", rn.endpoint.Spec.CableName,
+			rn.useIP)
+
+	case rn.endpoint.Spec.NATEnabled:
+		rn.useIP = rn.endpoint.Spec.PublicIP
+		rn.useNAT = rn.endpoint.Spec.NATEnabled
```

Comment:
```suggestion
		rn.useNAT = true
```
To match @tpantelis comments, and at this point rn.endpointSpec.NATEnabled has to be true... so..